#### Практическое задание: Урок 4. Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. Центральная предельная теорема

##### 1. Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800]. Найдите ее среднее значение и дисперсию.

In [1]:
import numpy as np

Вариант №1: Рассчитаем среднее и дисперсию по классической формуле:

In [4]:
# дано:
a = 200
b = 800

In [5]:
# среднее
Mx = (a+b)/2
Mx

500.0

In [6]:
# дисперсия
Dx = ((b-a)**2)/12
Dx

30000.0

Вариант №2: Сгенерируем случайную непрерывную величину A с равномерным распределением:

In [7]:
A = np.random.uniform(200, 800, (1, 20))
A

array([[624.57145149, 619.52321482, 729.45176733, 201.02796053,
        786.78102066, 385.53080117, 775.78680726, 672.39134556,
        588.40851075, 611.71932388, 306.67116902, 684.0491295 ,
        675.07665804, 751.80761187, 427.03297921, 759.05782829,
        462.80833702, 644.0231441 , 529.40320283, 219.78414094]])

In [8]:
# среднее
A.mean()

572.7453202141602

In [9]:
# дисперсия
A.var()

31333.755221824707

##### 2. О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

Возьмем формулу нахождения дисперсии и методом подбора находим правую границу:

In [10]:
Dx = 0.2
a = 0.5

In [19]:
b = 2.049
Dx = (x-0.5)**2/12
Dx

0.1999500833333333

Находим среднее значение по формуле:

In [20]:
Mx = (a+b)/2
Mx

1.2745

##### 3. Непрерывная случайная величина X распределена нормально и задана плотностью распределения 
f(x) = (1 / (4 * sqrt(2*pi))) *(exp(-((x+2)**2 / 32))). Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение)

Проверим соответствие заданной формулы классической формуле распределения вероятностей непрерывной случайной величины:

In [21]:
s = 4
C = 2*s**2
C

32

Следовательно:

In [22]:
Mx = a = 2

In [23]:
Dx = s**2
Dx

16

In [24]:
s

4

##### Ответ: а). M(X) = 2 б). D(X) = 16 в). std(X) (среднее квадратичное отклонение) = 4

##### 4. Рост взрослого населения города X имеет нормальное распределение. Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см. Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост: а). больше 182 см          б). больше 190 см            в). от 166 см до 190 см          г). от 166 см до 182 см          д). от 158 см до 190 см         е). не выше 150 см или не ниже 190 см         ё). не выше 150 см или не ниже 198 см           ж). ниже 166 см

In [25]:
Mx = 174
s = 8

а). больше 182 см

В соответствии с правилом трех сигм в интервале от -s до +s находится 68% значений. Рост больше 182 см находится в правом секторе за этими значениями. Следовательно:

In [26]:
P1 = (1 - 0.68)/2
P1

0.15999999999999998

б). больше 190 см

В соответствии с правилом трех сигм в интервале от -2s до +2s находится 95.4% значений.  Pост больше 190 см находится в правом секторе за значением +2s. Следовательно:

In [27]:
P2 = (1 - 0.954)/2
P2

0.02300000000000002

в). от 166 см до 190 см

В нашем случае рост 166 см - это граница сектора значений -s, а 190 см - +2s. Следовательно:

In [28]:
P3 = 0.68 + (0.954-0.68)/2
P3

0.817

г). от 166 см до 182 см

В соответствии с правилом трех сигм указанный интервал соответствует сектору значений от -s до +s. Следовательно:

In [29]:
P4 = 0.68

д). от 158 см до 190 см

Это интервал от -2s до +2s. Следовательно:

In [30]:
P5 = 0.954

е). не выше 150 см или не ниже 190 см

150 см - порог зачений -3s, 190 - +2s. Следовательно:

In [32]:
P6 = (1-0.9972)/2 + (1-0.954)/2
P6

0.024400000000000033

ё). не выше 150 см или не ниже 198 см

150 см - порог зачений -3s, 198 - +3s. Следовательно:

In [34]:
P7 = 1-0.9972
P7

0.0028000000000000247

ж). ниже 166 см

166 см - порог зачений левее -s. Следовательно:

In [35]:
P8 = (1-0.68)/2
P8

0.15999999999999998

##### 5. На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

D(X) равно сигма^2 (s^2). Следовательно:

In [36]:
Mx = 178
Dx = 25
s = Dx**(1/2)
s

5.0

In [37]:
# Mx + a*s = 190
a = (190 - Mx)/s
a

2.4

##### Ответ: рост человека, равный 190 см отклоняется на 2.4 сигмы от математического ожидания